Do not load calendar/addressbook plugins if not needed

rullzer opened the pull request:
Fixes #6711
Based on https://github.com/owncloud/core/pull/30149/commits/f0fb21cf3563088284c56be4eac48c7d9725cb21

Signed-off-by: Roeland Jago Douma <roeland@famdouma.nl>

rullzer committed:
Do not load calendar/addressbook plugins if not needed

Fixes #6711
Based on https://github.com/owncloud/core/pull/30149/commits/f0fb21cf3563088284c56be4eac48c7d9725cb21

Signed-off-by: Roeland Jago Douma <roeland@famdouma.nl>

MorrisJobke commented:
Does this also fix stuff like #8141?

georgehrke commented:
@MorrisJobke It should be as far as i can tell.

georgehrke commented:
I will check when testing this pr.

rullzer commented:
Yes it should

MorrisJobke reviewed (approved):
Tested, works and code makes sense 👍

rullzer committed:
When sharing calendars and addressbooks the principal has to be verified to be valid

https://github.com/owncloud/core/pull/30149/commits/d3fb8fcdd3a6b00bde0c3c9eb4039876e7fc1967

Signed-off-by: Roeland Jago Douma <roeland@famdouma.nl>

rullzer committed:
CalDAV and CardDAV plugins need to be registered for the principals collection as well

https://github.com/owncloud/core/pull/30149/commits/9f2e6431b88a9635b291b0e824abf74ba766616d

Signed-off-by: Roeland Jago Douma <roeland@famdouma.nl>

rullzer committed:
Adding repair step which cleans shares of invalid principals

https://github.com/owncloud/core/pull/30149/commits/edacf22fbce4deefc181c53dd84c5fb278a9a28d

Signed-off-by: Roeland Jago Douma <roeland@famdouma.nl>

rullzer committed:
Fix tests

Signed-off-by: Roeland Jago Douma <roeland@famdouma.nl>

MorrisJobke commented:
CI failure is due to a timeout :)

rullzer commented:
One more review please :)

rullzer commented:
Conflicts fixed and rebased.

MorrisJobke commented:
@georgehrke @blizzz @nickvergessen @ChristophWurst @skjnldsv Reviews would be nice :)

blizzz reviewed (changes requested):
the repair step might lose valid shares when

1. Nextcloud is in maintenance mode
2. AND external user backends are used

since apps are not loaded in this context.

rullzer commented:
@blizzz I can drop the repair job... but then we might have invalid shares....

blizzz commented:
> @blizzz I can drop the repair job... but then we might have invalid shares....

Or add a switch that prevents running in maintenance mode.

rullzer committed:
Move repairstep to a custom command

People that have issues can run it manually

Signed-off-by: Roeland Jago Douma <roeland@famdouma.nl>

rullzer commented:
@blizzz now they are only run if you run the expensive repair steps in case you run into it.

All our other repair steps are executed in maintenance mode so then it indeed would do :boom: in certain situations

blizzz commented:
> All our other repair steps are executed in maintenance mode so then it indeed would do :boom: in certain situations

Then it is better run as a seperate occ command, bound to apps, as i am afraid it would do more harm than good.

rullzer commented:
Nah during upgrade.

Now it is. occ maintenance:repair - - expensive

rullzer commented:
@blizzz ok one more time ;)

blizzz commented:
looks good now, apart from tests

MorrisJobke commented on lib/private/Repair.php:
```diff
@@ -47,6 +47,8 @@
 use OC\Template\JSCombiner;
 use OC\Template\SCSSCacher;
+use OCA\DAV\Connector\Sabre\Principal;
+use OCA\DAV\Repair\RemoveInvalidShares;
```
<img width="382" alt="bildschirmfoto 2018-05-31 um 17 22 21" src="https://user-images.githubusercontent.com/245432/40791200-33cde280-64f7-11e8-911d-a6a15be4bdd5.png">

🙈
